fairseq/modules/transformer_layer.py

```diff
@@ -187,44 +283,83 @@ def forward(
         # Note that we cannot use -inf here, because at some edge cases,
         # the attention weight (before softmax) for some padded element in query
         # will become -inf, which results in NaN in model parameters
-        if attn_mask is not None:
-            attn_mask = attn_mask.masked_fill(
-                attn_mask.to(torch.bool), -1e8 if x.dtype == torch.float32 else -1e4
+
+        if self.training:
+            self.ever_training = True
```

Comment:
Why have a variable that is the same as self.training?

Reply:
That is for the scenario where loading ckpt+continue training. We'd like to prevent this case to go fast path.